module/VuFind/src/VuFind/Controller/UpgradeController.php

```diff
@@ -821,6 +823,20 @@ public function getsourceversionAction()
         return $this->forwardTo('Upgrade', 'GetSourceDir');
     }
 
+    /**
+     * Organize and run critical, blocking checks
+     *
+     * @return string|null
+     */
+    protected function performCriticalChecks()
+    {
+        // Prevent the use of the deprecated blowfish cipher algo
+        $forward = $this->criticalCheckForBlowfishEncryption();
```

Comment:
I think performCriticalChecks should just `return $this->criticalCheckForBlowfishEncryption();` and then the calling code can do the null check and forward. This way, if we get more critical checks, we can just chain them here with `??`. This change would address the fact that right now, I don't think the return value from this method is getting handled correctly.